src/core/utils.js

```diff
@@ -7,6 +7,65 @@ import { hyperHTML } from "./import-maps.js";
 import { pub } from "./pubsubhub.js";
 export const name = "core/utils";
 
+export function getSpecTitle() {
+  return document.querySelector("h1#title");
+}
+
+const localizationStrings = {
+  en: {
+    default_title: "No Title",
+  },
+  de: {
+    default_title: "Kein Titel",
+  },
+};
+
+const l10n = getIntlData(localizationStrings);
+
+export function updateSpecTitleElem(conf) {
+  /** @type {HTMLElement} */
+  const specTitleElem =
+    document.querySelector("h1#title") || document.createElement("h1");
+
+  if (specTitleElem.isConnected) {
+    specTitleElem.remove();
+    if (specTitleElem.textContent.trim() === "") {
+      const msg =
+        'The document is missing a title, so using a default title. To fix this, please give your document a `<title>`. If you need special markup in the document\'s title, please use a `<h1 id="title">`.';
+      const msgTitle = "Document is missing a title";
+      showInlineError(specTitleElem, msg, msgTitle);
+    }
+  }
+
+  setDocumentTitle(conf, specTitleElem);
+
+  specTitleElem.id = "title";
+  specTitleElem.classList.add("title", "p-name");
+
+  return specTitleElem;
+}
+
+function setDocumentTitle(conf, specTitleElem) {
+  let documentTitle;
+
+  if (specTitleElem.textContent.trim() === "") {
+    specTitleElem.textContent = document.title || `${l10n.default_title}`;
+  }
+
+  documentTitle = norm(specTitleElem.textContent);
```

Comment:
Maybe just `let documentTitle` at this line?

Reply:
@saschanaz didn't get what you meant at first - thought it was change `documentTitle = norm(specTitleElem.textContent);` to `let documentTitle` - but @marcoscaceres explained to me what you meant. ^^ fixed, thanks!

Reply:
Oh, Marcos just read my mind 🤓